force-app/main/default/classes/DMLOperations.cls

```diff
@@ -160,36 +206,52 @@ public with sharing class DMLOperations {
 	 * @return The Account that was upserted.
 	 */
 	public static Account upsertAccount(String accountName) {
-		// Query for the Account
-		List<Account> accounts = [SELECT Id FROM Account WHERE Name = :accountName];
-
+        Account account;
+        // Query for the Account
+        List<Account> accounts = [SELECT Id, Name, Description FROM Account WHERE Name = :accountName LIMIT 1];
 		// if the Account exists, update the description
-
-		// otherwise, create a new Account and set the description
-
+        if (!accounts.isempty()) {
+            account = accounts[0];
+            account.Description = 'Updated Account';
+        }
+        // if the Account does not exist, create a new one
+        else {
+            account = new Account(Name = accountName, Description = 'New Account');
+        }
 		// Upsert the Account
-		return null;
+        upsert account;
+		return account;
 	}
-
+// can't seem to figure out why this test code fails or is skipped.
+//Test.startTest() keeps coming up. I made adjustments and spent time. Also asked in Slack. Might ask in office hours later this week.
     /**
      * Question 10
 	 * Manage the association between Contacts and Accounts
 	 * Ensure that for each Contact in the provided list, there is a corresponding Account Name that matches the Contact's last name.
 	 * If such an Account does not exist, the method creates it; if it does, the method ensures the Contact is associated with it. (Hint: use previous upsertAccount method - this may cause bad practices but for learning purposes it's okay.)
-     * Example Contact.LastName = 'Doe' Link to Account.Name = 'Doe' 
+     * Example Contact.LastName = 'Doe' Link to Account.Name = 'Doe'
      * Example Contact.LastName = 'Jane' Link to Account.Name = 'Jane'
 	 * @param contactsToUpsert List of Contacts to be upserted and linked to accounts.
 	 */
 	public static void upsertAccountsWithContacts(List<Contact> contactsToUpsert) {
 		// Loop through each Contact
-
+        for (Contact contact : contactsToUpsert) {
 		// Extract the Account Name from Contact's LastName
-
+            String accountName = contact.LastName;
 		// Get the account based on the name or create one if it doesn't exist
-
+        Account account;
+        try {
+            account = [SELECT Id, Name FROM Account WHERE Name = :accountName LIMIT 1];
+        }
+        catch (System.QueryException e) {
+            account = new Account(Name = accountName);
+            upsert account;
```

Comment:
Hint: use previous upsertAccount method - this may cause bad practices but for learning purposes it's okay.)

Try to see if you can use the method above that's already doing the same thing.